<launch>
<!-- Launch file for Livox MID360 LiDAR -->

	<arg name="rviz" default="false" />

	<param name="lidar_freq" type="int" value="10"/>
	<param name="frame_pub_count" type="int" value="10"/>
	<param name="frame_skip" type="int" value="0"/>
	<param name="filter_size" type="double" value="0.2" />
	<param name="filter_ceil" type="double" value="2.5" />
	<param name="filter_floor" type="double" value="0.1" />
	<param name="filter_dist" type="double" value="20.0" />
	<param name="filter_pc_size_each_pub" type="int" value="85" />
	<param name="pub_freq" type="int" value="10" />
	<param name="rviz_vis_flag" type="bool" value="true" />

	<node pkg="fast_lio" type="fy_series_publisher" name="fy_series_publisher" output="screen" /> 

	<group if="$(arg rviz)">
	<node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find fast_lio)/rviz_cfg/loam_livox.rviz" />
	</group>

</launch>
